File: tests/code-intelligence/parser/fixtures/sample.go
package main

func Greet(name string) string {
    return "Hello, " + name + "!"
}

type UserService struct {
    users map[string]User
}

func (s *UserService) GetUser(id string) (*User, error) {
    user, ok := s.users[id]
    if !ok {
        return nil, nil
    }
    return &user, nil
}

File: tests/code-intelligence/parser/fixtures/sample.py
def greet(name: str) -> str:
    """Greet a user by name."""
    return f"Hello, {name}!"

class UserService:
    def __init__(self):
        self._users = {}

    async def get_user(self, user_id: str):
        return self._users.get(user_id)

    @property
    def user_count(self) -> int:
        return len(self._users)
